package.json

```diff
@@ -4,7 +4,7 @@
   "description": "Spoke",
   "main": "src/server",
   "engines": {
-    "node": "6.10.3",
+    "node": ">=6.10.3",
```

Comment:
This should be `8.10` and not just anything -- otherwise e.g. node 10 will break lambda

Reply:
I'm not super familiar with the Lambda node runtime -- would anything that's not an exact match cause Lambda deployments to fail?

My thought is that we shouldn't _disallow_ folks from running Spoke with the node version of their choosing, which provides flexibility for those who want to take advantage of improvements in newer runtimes.

Reply:
Yes, unfortunately that's how it works.
https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html
(yes, custom runtimes could in theory open other possibilities, but I think easy setup is an important goal/target for this project)

Reply:
So lambda actually picks up on the engine value from `package.json`? Or is that maybe a claudia thing? I just took a quick look but I don't see anything about package.json in the Lambda runtime docs.

Reply:
Probably a claudia thing -- either way, I'm with @harpojaeger -- I think being specific is better because then it's a real target for devs to code within and for deployers/maintainers to depend on.  Otherwise, when do we say "no you can't use X because it's not supported" -- never.  But then deployers have no idea what to target or have to use the bleeding edge -- often not supported as widely in hosting contexts.

Reply:
I haven't been able to reproduce the issue with claudia failing to build when the node version is not a pinned version. I don't have an AWS account handy for testing lambda unfortunately, but I can't find anything to indicate unpinning could cause problems in lambda either.

I think avoiding pinning actually enhances security, since users and managed services won't have to fight tooling to consume critical security patches to their runtimes without first modifying the engine configuration in source/upstream.

See more discussion here on security and support: https://github.com/spokecommunity/Spoke/pull/5#discussion_r267599167

I did go ahead and add an upper version limit so that only LTS node releases will match. If we don't already, then I think clearly documenting version support policy in a more human-centric format would be useful.

Reply:
I'm not saying that it will fail to build if the version is not pinned.  However, if we do not pin the version, then people will inevitably write code that requires version 9, and then people running an environment with node version 8 will be unable to run.

++ to documenting it in the readme and/or contributing that we are currently targeting version 8.10.  I still think we should have something pinned in the package.json, but maybe I'm not understanding what happens if we have a pinned version -- does that mean node version 9 will refuse to run it if it says '`"node": "8.10"` (or 6.10)?

Reply:
> what happens if we have a pinned version -- does that mean node version 9 will refuse to run it if it says '"node": "8.10" (or 6.10)?

No, node and npm (at least up to node 10 and npm 5 so far) do not pay attention to engines at runtime. The engines section of package.json was introduced only in order to make sure that your stuff could work with other stuff when consumed as an npm dependency, so most tools don't care about the top level package. Yarn is an exception, and will exit with an error if you run package scripts or try to install while your node runtime isn't a match for the engines config.

> However, if we do not pin the version, then people will inevitably write code that requires version 9, and then people running an environment with node version 8 will be unable to run.

This is possible today due to the way engines config works, and if our test automation strategies don't already account for this then we should work on that. 😄The docker updates in this PR should make testing multiple node versions pretty simple. [Travis build matrices](https://docs.travis-ci.com/user/build-matrix/) are awesome for this!